tests/unittests/sources/test_aliyun.py

```diff
@@ -179,6 +179,44 @@ def test_with_mock_server(self, m_is_aliyun, m_resolv):
             "metadata (http://100.100.100.200)", self.ds.subplatform
         )
 
+    @mock.patch("cloudinit.sources.DataSourceEc2.util.is_resolvable")
+    @mock.patch("cloudinit.sources.DataSourceAliYun._is_aliyun")
+    @mock.patch("cloudinit.net.find_fallback_nic")
+    @mock.patch("cloudinit.net.ephemeral.maybe_perform_dhcp_discovery")
+    @mock.patch("cloudinit.sources.DataSourceEc2.util.is_FreeBSD")
+    def test_aliyun_local_with_mock_server(
```

Comment:
When running locally, I get:
`tests/unittests/sources/test_aliyun.py::TestAliYunDatasource::test_aliyun_local_with_mock_server - FileNotFoundError: [Errno 2] No such file or directory: '/sys/class/net/eth0/operstate'`

Did you forget a mock?